lib/src/pool.rs

```diff
@@ -14,8 +15,11 @@ pub struct ConnectionManager {
 }
 
 impl ConnectionManager {
-    pub fn new(uri: &str, user: &str, password: &str) -> Result<Self> {
-        let info = ConnectionInfo::new(uri, user, password)?;
+    pub fn new(uri: &str, user: &str, password: &str, client_certificate: Option<&ClientCertificate>) -> Result<Self> {
+        let mut info = ConnectionInfo::new(uri, user, password)?;
+        if let Some(client_certificate) = client_certificate {
+            info.with_client_certificate(client_certificate);
+        }
```

Comment:
Without the `with` this would then be

```suggestion
        let info = ConnectionInfo::new(uri, user, password, client_certificate)?;
```